### PR TITLE
fix: passing `BindingResolveOptions#alias` from js to rust with maintaining order

### LINKED ISSUE #790: Passing `BindingResolveOptions#alias` from js to rust with maintaining order
https://github.com/rolldown/rolldown/blob/c304a0dcde43234654626f8695cd745f37de4d33/crates/rolldown_binding/src/options/binding_input_options/binding_resolve_options.rs#L8-L11

### Use `Option<Vec<(String, Vec<String>)>>>`

The problem is that NAPI-RS so far doesn't support tuples

### Define a temporary struct like

```rust
pub struct AliasItem {
  pub name: String,
  pub paths: Vec<String>,
}
```

and write `Option<Vec<AliasItem>>>`.

---

I personally prefer using tuples on this case. However this requires [discussing with NAPI-RS if this feature is really needed](https://github.com/napi-rs/napi-rs/issues/1372).

### PR DESCRIPTION
**Pull request by PengBoUESTC**
### Description

convert `alias` to `Vec<AliasItem>`

for issue https://github.com/rolldown/rolldown/issues/790

### CODE REVIEW

File: crates/rolldown_binding/src/types/binding_resolve_alias.rs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+use serde::Deserialize;
+
+#[napi_derive::napi(object)]
+#[derive(Deserialize, Debug, Default)]
+#[serde(rename_all = "camelCase")]
+pub struct AliasItem {
+  pub find: String,
+  pub replacement: Vec<String>,
```
Comment (hyf0) [inline crates/rolldown_binding/src/types/binding_resolve_alias.rs]:
```suggestion
  pub replacements: Vec<String>,
```

File: crates/rolldown_binding/src/types/binding_resolve_alias.rs
Comment (hyf0) [inline crates/rolldown_binding/src/types/binding_resolve_alias.rs:1]: filename should be `alias_item.rs`.

File: packages/rolldown/src/options/input-options.ts [hunk 2]
```diff
@@ -17,8 +21,12 @@ export interface InputOptions {
   shimMissingExports?: BindingInputOptions['shimMissingExports']
 }
 
+export type Alias = Omit<AliasItem, 'replacement'> & {
+  replacement: string
+}
+
 export type RolldownResolveOptions = Omit<BindingResolveOptions, 'alias'> & {
-  alias?: Record<string, string>
+  alias?: Record<string, string> | Alias[]
```
Comment (hyf0) [inline packages/rolldown/src/options/input-options.ts]: Please don't change the type of `alias`. For users, they still pass alias using object.

**Review by hyf0 — CHANGES_REQUESTED** [review]

File: packages/rolldown/src/options/input-options.ts (hunk 2 above)
Comment (PengBoUESTC) [reply packages/rolldown/src/options/input-options.ts]: But I notice that, in `vite` and `rollup`, config `alias` is support for Array and object

Comment (hyf0) [reply packages/rolldown/src/options/input-options.ts]: Yeah. Vite use https://github.com/rollup/plugins/tree/master/packages/alias#entries under the hood. The API it provided has different usages for `array` or `object`, which means you could not use `object` to [use `customResolver` API](https://github.com/rollup/plugins/blob/9e7f576f33e26d65e9f2221d248a2e000923e03f/packages/alias/README.md?plain=1#L162).

However, we don't have this need. In our cases, `object` and `array` does the same thing. To keep it simple, supporting `object` only would be fine.

Comment (PengBoUESTC) [reply packages/rolldown/src/options/input-options.ts]: Ok, get it

File: crates/rolldown_binding/src/types/binding_resolve_alias.rs (hunk 1 above)
Comment (PengBoUESTC) [reply crates/rolldown_binding/src/types/binding_resolve_alias.rs]: Done

File: crates/rolldown_binding/src/types/binding_resolve_alias.rs
Comment (PengBoUESTC) [reply crates/rolldown_binding/src/types/binding_resolve_alias.rs:1]: Done

**Review by hyf0 — APPROVED** [review]
Thanks

**Pull request merged**